bin/remove_empty_rows_and_columns.py

```diff
@@ -0,0 +1,101 @@
+"""
+Create a copy of CSV files in a directory with empty columns and rows removed.
+
+This is a program for use in development. It transforms source files by
+removing any columns with an empty heading and any empty rows before
+writing them to an output directory.
+"""
+
+import glob
+import os
+import logging
+import csv
+from argparse import ArgumentParser
+
+VERSION = '1.2.1-alpha'
+
+
+def main():
+    """Create a copy of CSV files in a directory with empty columns and rows removed."""
+    parser = ArgumentParser(
+        description='Create a copy of CSV files in a directory with empty columns and rows removed.',
+        epilog=f'Version: {VERSION}')
+
+    parser.add_argument('-i', '--input-dir',
+                        type=str,
+                        required=True,
+                        help='Input directory containing CSV files')
+
+    parser.add_argument('-o', '--output-dir',
+                        type=str,
+                        required=True,
+                        help='Output directory to write modified files')
+
+    args = parser.parse_args()
+
+    logging.basicConfig(
+        format='t=%(asctime)s lvl=%(levelname)s msg=%(message)s', level='INFO')
+
+    for directory in (args.input_dir, args.output_dir):
+        if not os.path.isdir(directory):
+            raise ValueError(
+                f'{directory} does not exist or is not a directory')
+
+    for filename in glob.glob(os.path.join(args.input_dir, '*.csv')):
+        basename = os.path.basename(filename)
+        out_filename = os.path.join(args.output_dir, basename)
+        with open(out_filename, 'w', encoding='utf-8-sig') as outfile:
+            with open(filename, newline='', encoding='utf-8-sig') as infile:
+                reader = csv.reader(infile)
+                headings = next(reader)
+                removeColumnsAfterIndex = len(headings)
```

Comment:
Should really be called `removeColumnsFromIndex`? i.e. my understanding is that columns at index and above are removed not the ones "after" index?

Reply:
Agreed.